File: src/main/java/dev/verloren/midnight/CompactFileType.java
package dev.verloren.midnight;

import com.intellij.openapi.fileTypes.LanguageFileType;
import dev.verloren.midnight.icons.MidnightIcons;
import org.jspecify.annotations.NonNull;

import javax.swing.*;

/**
 * Represents the Compact source file type ({@code .compact}) in IntelliJ IDEA.
 *
 * <p>Binds the file extension to {@link CompactLanguage#INSTANCE} and provides
 * file metadata, descriptions, and the Midnight file icon displayed in the project tree
 * and editor tabs.</p>
 */
public class CompactFileType extends LanguageFileType {

  public static final CompactFileType INSTANCE = new CompactFileType();

  private CompactFileType() {
    super(CompactLanguage.INSTANCE);
  }

  @Override
  public @NonNull String getName() {
    return "Compact";
  }

  @Override
  public @NonNull String getDescription() {
    return "Compact language file";
  }

  @Override
  public @NonNull String getDefaultExtension() {
    return "compact";
  }

  @Override
  public Icon getIcon() {
    return MidnightIcons.FILE;
  }
}

File: src/main/kotlin/dev/verloren/midnight/CompactFileType.kt
package dev.verloren.midnight

import com.intellij.openapi.fileTypes.LanguageFileType
import dev.verloren.midnight.icons.MidnightIcons
import javax.swing.Icon

class CompactFileType : LanguageFileType(CompactLanguage) {

    override fun getName() = "Compact"

    override fun getDescription() = "Compact language file"

    override fun getDefaultExtension() = "compact"

    override fun getIcon(): Icon = MidnightIcons.FILE

}

@JvmField
val INSTANCE = CompactFileType()